internal/cfgmgr/cfgmgr.go

```diff
@@ -0,0 +1,149 @@
+	CM_GETIDLIST_FILTER_POWERRELATIONS     = uint64(0x00000010)
+	CM_GETIDLIST_FILTER_BUSRELATIONS       = uint64(0x00000020)
+	CM_GETIDLIST_DONOTGENERATE             = uint64(0x10000040)
+	CM_GETIDLIST_FILTER_TRANSPORTRELATIONS = uint64(0x00000080)
+	CM_GETIDLIST_FILTER_PRESENT            = uint64(0x00000100)
+	CM_GETIDLIST_FILTER_CLASS              = uint64(0x00000200)
+	CM_GETIDLIST_FILTER_BITS               = uint64(0x100003FF)
+
+	CM_LOCATE_DEVNODE_NORMAL       = uint64(0x00000000)
+	CM_LOCATE_DEVNODE_PHANTOM      = uint64(0x00000001)
+	CM_LOCATE_DEVNODE_CANCELREMOVE = uint64(0x00000002)
+	CM_LOCATE_DEVNODE_NOVALIDATION = uint64(0x00000004)
+
+	DEVPROP_TYPE_STRING        = uint64(0x00000012)
+	DEVPROP_TYPEMOD_LIST       = uint64(0x00002000)
+	DEVPROP_TYPE_STRING_LIST   = uint64(DEVPROP_TYPE_STRING | DEVPROP_TYPEMOD_LIST)
+	DEVPKEY_LOCATIONPATHS_GUID = "a45c254e-df1c-4efd-8020-67d146a850e0"
+
+	STRING_BUFFER_SIZE = uint64(1024)
+)
+
+func getDevPKeyDeviceLocationPaths() (*devPropKey, error) {
+	guid, err := guid.FromString(DEVPKEY_LOCATIONPATHS_GUID)
+	if err != nil {
+		return nil, err
+	}
+	return &devPropKey{
+		fmtid: guid,
+		pid:   37,
+	}, nil
+}
+
+func GetDeviceLocationPathsFromIDs(ids []string) ([]string, error) {
+	result := []string{}
+	devPKeyDeviceLocationPaths, err := getDevPKeyDeviceLocationPaths()
+	if err != nil {
+		return nil, err
+	}
+	for _, id := range ids {
+		var devNodeInst uint32
+		err = cmLocateDevNode(&devNodeInst, id, CM_LOCATE_DEVNODE_NORMAL)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to locate device node for %s", id)
+		}
+		propertyType := uint64(0)
+		propertyBufferSize := STRING_BUFFER_SIZE
+		var propertyBuffer [STRING_BUFFER_SIZE]uint16
+		err = cmGetDevNodeProperty(devNodeInst, devPKeyDeviceLocationPaths, &propertyType, &propertyBuffer[0], &propertyBufferSize, 0)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to get location path property from device node for %s with", id)
+		}
+		if propertyType != DEVPROP_TYPE_STRING_LIST {
+			return nil, fmt.Errorf("expected to return property type DEVPROP_TYPE_STRING_LIST %d, instead got %d", DEVPROP_TYPE_STRING_LIST, propertyType)
+		}
+		if propertyBufferSize > STRING_BUFFER_SIZE {
+			return nil, fmt.Errorf("location path is too long, size %d", propertyBufferSize)
+		}
+		formattedResult := convertFirstNullTerminatedValueToString(propertyBuffer[:propertyBufferSize])
+		result = append(result, formattedResult)
+	}
+
+	return result, nil
+}
+
+// helper function that finds the first \u0000 rune and returns the wide string as a regular go string
+func convertFirstNullTerminatedValueToString(buf []uint16) string {
+	r := utf16.Decode(buf)
+	converted := string(r)
+	zerosIndex := strings.IndexRune(converted, '\u0000')
+	if zerosIndex != -1 {
+		converted = converted[:zerosIndex]
+	}
+	return converted
+}
+
+func GetChildrenFromInstanceIDs(parentIDs []string) ([]string, error) {
```

Comment:
Would it be more useful to return a mapping from parentID -> childID, rather than all the children together in a list?

Reply:
Could you explain why? 

The goal of returning just all the children together is that we don't necessarily care what the parentID is after we get the actual target devices (iow a GPU DDA'd onto the pod). We just need the target devices' information moving forward. Does that make sense?

Reply:
I don't know the use cases, so I was asking if it would be useful to know the parent -> child mapping in some cases. From your reply it sounds like it's not needed for GPU scenarios. I would consider if there could be other cases in the future where that mapping would be helpful to have.

Reply:
I'm going to think about this some more. For now it is not needed.